### PR TITLE
[core] - chore(core): favor api keys in data sources

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at using core API keys over Dust's keys in datasource blocks (c.f. https://github.com/dust-tt/tasks/issues/6750)

## Risk

We are dealing with secrets so let's make sure they don't leak.

## Deploy Plan

Create secrets in k8s -> deploy `core`

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
Code LGTM.

I'd like to not make a call to env::var on each data source interaction which is "blocking" and not completely trivial as it likely goes through the OS boundary.

A small utils thing that caches the value of an env in a singleton per instance would do the trick.

Maybe this is an overkill. We might want to look at the documentation to convince ourselves? Maybe that function already does it?

**Comment by JulesBelveze** [comment]
@spolu Seems like `env::var` fetches the variable at runtime. 

But this should this do trick I guess? https://doc.rust-lang.org/std/macro.env.html

Otherwise I'll go for a small impl 😃

**Comment by spolu** [comment]
Huh... interesting that sounds dangerous but kind of makes sense. Let's use that then :+1:

**Comment by spolu** [comment]
Can you refactor all use of `std::env::var` in core (wrapped in tokio non blocking thingy to avoid blocking tokio workers while going to the OS to just this macro?

**Comment by spolu** [comment]
So we're not built on the production machines so that does not work.

**Comment by JulesBelveze** [comment]
Will go for the a util tool then 👍🏼

**Comment by spolu** [comment]
Let's simply wrap the tokio machinery in an async get_env for now :+1:

**Comment by spolu** [comment]
https://dust.tt/w/0ec9852c2f/assistant/d28abffc56

**Comment by JulesBelveze** [comment]
As per IRL removed the `spawn_blocking`

File: core/src/providers/mistral.rs [hunk 1]
```diff
@@ -796,19 +796,23 @@ impl LLM for MistralAILLM {
     fn id(&self) -> String {
         self.id.clone()
     }
-
     async fn initialize(&mut self, credentials: Credentials) -> Result<()> {
```
Comment (fontanierh) [inline core/src/providers/mistral.rs:799]: Seems like we don't need it to be async then ?

File: core/src/providers/openai.rs [hunk 2]
```diff
@@ -1415,17 +1415,23 @@ impl LLM for OpenAILLM {
     }
 
     async fn initialize(&mut self, credentials: Credentials) -> Result<()> {
-        match credentials.get("OPENAI_API_KEY") {
-            Some(api_key) => {
-                self.api_key = Some(api_key.clone());
+        // Give priority to `CORE_DATA_SOURCES_OPENAI_API_KEY` env variable
```
Comment (fontanierh) [inline core/src/providers/openai.rs:1418]: same comment wrt asyncness of the method

Comment (spolu) [reply core/src/providers/openai.rs:1418]: Yes but that's a broader refactor as this method is part of the signature of the trait above

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**